python/ray/rllib/models/modelv2.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from ray.rllib.models.model import restore_original_dimensions
+
+
+class ModelV2(object):
+    """Defines a Keras-style abstract network model for use with RLlib.
+
+    This interface is used for both TF and Torch custom models. Experimental.
+
+    Attributes:
+        obs_space (Space): observation space of the target gym env. This
+            may have an `original_space` attribute that specifies how to
+            unflatten the tensor into a ragged tensor.
+        action_space (Space): action space of the target gym env
+            num_outputs (int): the size of the output vector of the model
+        num_outputs (int): the size of the output vector of the model
+        options (dict): options for the model, documented in ModelCatalog
+    """
+
+    def __init__(self, obs_space, action_space, num_outputs, options):
+        """Initialize the model.
+
+        This method should create any variables used by the model.
+        """
+
+        self.obs_space = obs_space
+        self.action_space = action_space
```

Comment:
Should the model take in the environment instead of the observation/action spaces. Maybe in some cases the environments have some attributes that are needed in the model. Not sure about this, but just a thought.